packages/lib/domainManager/deploymentServices/cloudflare.ts

```diff
@@ -134,7 +138,7 @@ export const deleteDnsRecord = async (domain: string) => {
       });
     }
 
-    return searchResult.result[0];
+    return searchResult.result[0] as (typeof searchResult.result)[0] | null;
```

Comment:
It was the reason for the error during deletion when the result could be undefined.